include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -330,6 +330,8 @@ struct __merge_sort_global_submitter<_IndexT, __internal::__optional_kernel_name
         {
         }
     };
+    template <typename Rng>
+    DropViews(Rng&, const WorkDataArea&) -> DropViews<Rng>;
```

Comment:
Do we want to leave a comment explaining why the deduction guide is necessary to workaround a compiler issue?

Reply:
Yes, that would be helpful.  I've left a comment about the reason to add the deduction guide.

However, I still do not understand whether this is a fix of oneDPL issue (relying on template argument deduction in a context where it is not expected to work), or a compiler issue for a better comment.

Reply:
I've updated the comment. It is indeed a clang++ bug.